### PR TITLE
BUG: fix rolling with centering and axis=1 (#46135)

### PR DESCRIPTION
**Pull request by jyuv**
- [x] closes #46135
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by jyuv** [comment]
@github-actions pre-commit

File: pandas/tests/window/test_rolling.py [hunk 1]
```diff
@@ -1781,3 +1781,21 @@ def test_step_not_integer_raises():
...
+
+    expected = DataFrame(
+        {
+            "a": [None, None, None, None, None, None],
+            "b": [3, 1, 0, 2, 0, 2],
+            "c": [None, None, None, None, None, None],
+        }
+    )
+
+    tm.assert_frame_equal(result, expected, check_dtype=False)
```
Comment (mroeschke) [inline pandas/tests/window/test_rolling.py:1801]: Could you build the expected DataFrame with the correct dtype so you don't have to specify `check_dtype=False`?

Comment (jyuv) [reply pandas/tests/window/test_rolling.py]: Just Did :)

**Comment by jyuv** [comment]
@github-actions pre-commit

File: pandas/tests/window/test_rolling.py [hunk 2]
```diff
@@ -1781,3 +1781,17 @@ def test_step_not_integer_raises():
 def test_step_not_positive_raises():
     with pytest.raises(ValueError, match="step must be >= 0"):
         DataFrame(range(2)).rolling(1, step=-1)
+
+
+def test_rolling_center_axis_1():
```
Comment (mroeschke) [inline pandas/tests/window/test_rolling.py:1786]: Could you move to `test_win_type.py`?

**Comment by jyuv** [comment]
@github-actions pre-commit

**Comment by jyuv** [comment]
Hi,
maybe I'm missing something but it seemed to me that the failures in the checks are unrelated to the commited changes.
Can someone please take a look at it?

**Comment by mroeschke** [comment]
You will need to merge main and push to this branch again. Looks like previously the CI didn't run all the jobs

File: doc/source/whatsnew/v1.5.0.rst [hunk 3]
```diff
@@ -474,6 +474,7 @@ Performance improvements
 
 Bug fixes
 ~~~~~~~~~
+- Fixed bug in window rolling when centering with axis=1 (:issue:`46135`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.5.0.rst:477]: This should go under the `Groupby/resample/rolling` section

Comment (mroeschke) [reply doc/source/whatsnew/v1.5.0.rst:477]: Also should mention specifically when `center=True` and `win_type` is specified.

**Review by mroeschke — COMMENTED** [review]
Thanks, looks pretty good. Just needs a small adjustment in the whatsnew note.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jyuv

**Pull request merged**